### PR TITLE
[pdata] Remove Val suffix from pcommon.Value getter/setter methods

### PR DESCRIPTION
**Pull request by dmitryax**
Resolves https://github.com/open-telemetry/opentelemetry-collector/issues/6081

### CODE REVIEW

**Comment by bogdandrutu** [comment]
I like the new API... Want to hear more opinions. @open-telemetry/collector-approvers ?

**Comment by dmitryax** [comment]
@bogdandrutu If we merge this, we will still have `(Set)[Double|Int]Val` methods on `pmetric.NumberDataPoint` and `pmetric.Examplar` structs, but I believe it's good to keep the `Val` suffix there because they have other unrelated fields and they look through another _Value_ entity, so we should keep something related to _value_ in those method names.

**Comment by bogdandrutu** [comment]
Maybe that should be Value?

**Comment by bogdandrutu** [comment]
@dmitryax what about the metrics data type then? We definitely are not consistent here

**Comment by tigrannajaryan** [comment]
I am not sure this is an improvement. Get prefix is explicitly not recommended by Effective Go for getters. Granted, Val suffix is also not pretty.

Minor argument in favour of Val suffix: it is faster to use with autocomplete, less characters to type until you get the right hint.

However, I can live with either approach.

**Review by TylerHelmuth — APPROVED** [review]
I like this a lot. Get and Set feel like natural prefixes for this situation, even if it goes against Go naming standards

**Comment by dmitryax** [comment]
> Maybe that should be Value?

Yes, `(Set)[Double|Int]Value` on `pmetric.NumberDataPoint` and `pmetric.Examplar` sounds better to me 👍 

> @dmitryax what about the metrics data type then? We definitely are not consistent here

We are not, yes. We don't have any prefix or suffix for the metric type getters, it's just `Sum()`, `Gauge()` etc. It goes through `Data` oneof protobuf messages called `Data`. I'm not sure what we can do there to bring consistency.  Making it `SumValue` or `SumData` doesn't feel convenient.

**Comment by TylerHelmuth** [comment]
> We definitely are not consistent here 

This is pretty annoying, and if we can't use `Get|Set` everywhere then I'd _probably_ vote for consistency over anything else.  But if I was a new user to the pdata API, I think `GetString` and `SetString` are nicer than `StringVal` and `SetStringVal`.  

But I also have a bias towards languages with Getters and Setters.  An argument could be made that for Go `Get` isn't the expected name and would throw users off.

**Comment by TylerHelmuth** [comment]
>  it's just Sum(), Gauge()

This makes me really wish we could just do `SetString` and `String`.  `GetSum` and `GetGauge` don't feel right.  (Which is maybe a good argument against using `Get` for `Value`.)

**Comment by bogdandrutu** [comment]
@dmitryax please summarize the conversation in the issue:
* 4 places where we have this pattern right now (please enumerate them, and add links so people can find them easily), and all somehow inconsistent for naming.
* Let's play with the proposals: 1. use get/set (see how that impact other places); 2. In case of Metric/Exemplar should we have `SumData` to be consistent with `DoubleValue`?

So let's enumerate the rules that we want to apply (or exceptions if any) and apply them and see what we get. By apply just some code snippets it is fine.

**Review by jpkrohling — APPROVED** [review]
I preferred `BoolVal` to `GetBool` until I realized that `Get` would be a better pair to `Set` in `SetBool`.

**Comment by dmitryax** [comment]
Please do not merge until a decision is made on https://github.com/open-telemetry/opentelemetry-collector/issues/6081#issuecomment-1258364386

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
To avoid merging until decision in the issue is made

File: exporter/loggingexporter/internal/otlptext/databuffer.go
Comment (bogdandrutu) [inline exporter/loggingexporter/internal/otlptext/databuffer.go:275]: `ValueTypeStr`?

File: CHANGELOG.md
Comment (bogdandrutu) [inline CHANGELOG.md:40]: Update?

File: exporter/loggingexporter/internal/otlptext/databuffer.go
Comment (bogdandrutu) [inline exporter/loggingexporter/internal/otlptext/databuffer.go:287]: This lacks the ValueTypeBytes? I will take care of this in a separate PR.

Comment (dmitryax) [reply exporter/loggingexporter/internal/otlptext/databuffer.go:287]: Yes, nice catch.

File: CHANGELOG.md
Comment (dmitryax) [reply CHANGELOG.md]: updated

File: exporter/loggingexporter/internal/otlptext/databuffer.go
Comment (dmitryax) [reply exporter/loggingexporter/internal/otlptext/databuffer.go]: renamed

File: pdata/pcommon/common.go
Comment (bogdandrutu) [inline pdata/pcommon/common.go:208]: Consider to have this comment on the "SetStr" as well.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
